rust-tdshim/src/mp.rs

```diff
@@ -224,23 +231,23 @@ pub fn mp_accept_memory_resource_range(cpu_num: u32, address: u64, size: u64) {
         }
     }
 
-    wait_for_ap_arrive(cpu_num);
+    wait_for_ap_arrive(active_ap_cnt);
 
-    let mut stacks: Vec<u8> = Vec::with_capacity(0x1000 * (cpu_num as usize));
+    let mut stacks: Vec<u8> = Vec::with_capacity(AP_TEMP_STACK_SIZE * active_ap_cnt as usize);
     let mut mail_box = MailBox::new(get_mem_slice_mut(SliceType::MailBox));
 
     // BSP calles the same function parallel_accept_memory to accept memory,
     // so set the firmware arguments here.
     // To do: Set these parameter only in ap_assign_work() when there's
     // multiple cpus.
-    mail_box.set_fw_arg(1, cpu_num as u64);
+    mail_box.set_fw_arg(1, active_ap_cnt as u64 + 1);
     mail_box.set_fw_arg(2, address + align_low);
     mail_box.set_fw_arg(3, address + size);
 
     if major_part > 0 {
-        for i in 1..cpu_num {
-            let ap_stack = stacks.as_ptr() as u64 + i as u64 * 0x800;
-
+        // 0 is the bootstrap processor running this code
```

Comment:
Thanks. There are a couple of other occurences of `1..` so for maximum safety, you could do something like this to encapsulate the hard-coded offset:

```rust
use std::ops::Range;

fn make_apic_range(end: usize) -> Range<usize> {
    // 0 is the bootstrap processor running this code
    let start = 1;

    Range { start, end }
}

// ...

for i in make_apic_range(cpu_num) {
    // ...
}

for i in make_apic_range(active_ap_cnt) {
    // ...
}

for i in make_apic_range(ap_num) {
    // ...
}
```

Reply:
It's a good idea, thanks for your advice😊